src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -225,7 +225,11 @@ public static ZipArchive ReadZipFile (string filename, bool strictConsistencyChe
 				files.Add (outfile);
 				var dt = File.Exists (outfile) ? File.GetLastWriteTimeUtc (outfile) : DateTime.MinValue;
 				if (forceUpdate || entry.ModificationTime > dt) {
-					entry.Extract (destination, fullName, FileMode.OpenOrCreate);
+					try {
+						entry.Extract (destination, fullName, FileMode.OpenOrCreate);
+					} catch (PathTooLongException) {
+						throw new PathTooLongException ($"Could not extract \"{fullName}\" to \"{outfile}\". Path is too long.");
```

Comment:
Should we mention what `.zip` file we're extracting from?

Otherwise we're going to get things like:

    Could not extract "bin/classes.jar" to "C:\really\long\path\bin\classes.jar"

"Better," because of our new use of short filenames, the "long path" in the above message will be something like `lp\1\jl\bin\classes.jar`, which is inscrutable for knowing what went wrong and/or fixing it.